### PR TITLE
In-context-add-options feature | Topology

### PR DESCRIPTION
**Pull request by hemantsaini-7**
**Description:**

- Topology-In-context-add-options feature | Topology-automation

**Jira Id:**
     - [ODC-6672](https://issues.redhat.com/browse/ODC-6672)

**Pre-conditions/setup:**

**Checks for approving Epic scenarios Automation PR:**

- [x] Execute the @to-do tagged gherkin scripts manually
- [x] Convert the @to-do gherkin scripts to cypress automation scripts
- [x] Once scripts are automated, replace tag @to-do with @epic-number
- [x] Execute the scripts in Remote cluster

**Refactoring criteria for approving Automation PR:**

- [ ] update the test cases count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands:**
Example:
```
    export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
    BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
    BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.11-042801.devcluster.openshift.com/
    export BRIDGE_KUBEADMIN_PASSWORD
    export BRIDGE_BASE_ADDRESS
    oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
    oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
    oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
    ./test-cypress.sh -p topology
```
**Screen shots**:

topology-context-add-options.feature
image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by SparshKesari** [comment]
/lgtm

All tests are green and working. 

image

**Review by SparshKesari — APPROVED** [review]

File: frontend/packages/topology/integration-tests/support/page-objects/topology-add-options-po.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const topologyAddOptionsPO = {
+  samples: '[data-test-action="Samples"]',
+  importFromGit: '[data-test-action="Import from Git"]',
+  containerImage: '[data-test-action="Container Image"]',
+  catalog: '[data-test-action="From Catalog"]',
+  database: '[data-test-action="Database"]',
+  operatorBacked: '[data-test-action="Operator Backed"]',
+  helmCharts: '[data-test-action="Helm Charts"]',
+  eventsource: '[data-test-action="Event Source"]',
+  channel: '[data-test-action="Channel"]',
```
Comment (sanketpathak) [inline frontend/packages/topology/integration-tests/support/page-objects/topology-add-options-po.ts:10]: Make a dynamic function for it and use it

Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/support/page-objects/topology-add-options-po.ts:10]: Like this [data-test-action="${actionName}"]

Comment (hemantsaini-7) [reply frontend/packages/topology/integration-tests/support/page-objects/topology-add-options-po.ts]: updated

File: frontend/packages/topology/integration-tests/features/topology/topology-in-context-add-options.feature [hunk 2]
```diff
@@ -11,22 +11,23 @@ Feature: Add in context from the Developer Catalog
...
 
-        @smoke @to-do
+        @smoke
         Scenario: Add to Project in Context options: T-10-TC01
-             When user right clicks on graph
-              And user clicks on Add to Project
-             Then user can see in context options Samples, Import from Git, Container Image, From Catalog, Database, Operator Backed, Helm Chart, Event Source, Channel
+             When user right clicks on empty graph view
+              And user hovers on Add to Project
+             Then user can see in context options "Samples", "Import from Git", "Container Image", "From Catalog", "Database", "Operator Backed", "Helm Chart", "Event Source", "Channel"
```
Comment (sanketpathak) [inline frontend/packages/topology/integration-tests/features/topology/topology-in-context-add-options.feature:18]:
```suggestion
             Then user can see in context options "Samples", "Import from Git", "Container Image", "From Catalog", "Database", "Operator Backed", "Helm Charts", "Event Source", "Channel"
```

**Comment by hemantsaini-7** [comment]
Pr only affects E2E tests
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**